File: day_01/day_01.clj
(ns day_01
  (:require [clojure.java.io :as io]))

(defn -main []
  (let [list1 (atom [])
        list2 (atom [])
        ans (atom 0)]
    (with-open [reader (io/reader "data.txt")]
      (doseq [line (line-seq reader)]
        (let [[num1 num2] (map (fn [part] (Integer/parseInt part))
                               (clojure.string/split line #"\s+"))]
          (swap! list1 conj num1)
          (swap! list2 conj num2))))
    (reset! list1 (sort @list1))
    (reset! list2 (sort @list2))
    (doseq [[num1 num2] (map vector @list1 @list2)]
      (swap! ans + (Math/abs (- num1 num2))))
    (println @ans)))

File: day_01/Day01.java
import java.io.*;
import java.util.*;
import java.lang.Math;

public class Day01 {
	public static void main(String[] args) {
		List<Integer> list1 = new ArrayList<>();
		List<Integer> list2 = new ArrayList<>();

		try (BufferedReader reader = new BufferedReader(new FileReader("data.txt"))) {
			String line;

			while ((line = reader.readLine()) != null) {
				String[] parts = line.trim().split("\\s+");

				list1.add(Integer.parseInt(parts[0]));
				list2.add(Integer.parseInt(parts[1]));
			}
			
		} catch (IOException e) {
			e.printStackTrace();
		}

		Collections.sort(list1);
		Collections.sort(list2);

		long ans = 0;
		for (int i = 0; i < list1.size(); ++i) {
			ans += Math.abs(list1.get(i) - list2.get(i));
		}

		System.out.println(ans);
	}
}
